README
-------------------------------------
PyLan Manager
-------------------------------------

See INSTALL file for instruction 


ChangeLog
commit b188e05b897360aa135212c43023ddb36f1690ae
Author: Fernando Henrique <liquuid@liquuid.net>
Date:   Sun Feb 21 19:33:41 2010 -0300

    script para testes , agindo como o pylan, mas com tempo reduzido

commit 50fa3e463aa98d495207a706c04ecf874bc0f2a5
Author: Fernando Henrique <liquuid@liquuid.net>
Date:   Sun Feb 21 17:53:06 2010 -0300

    Revert "adicionando configuações para o VIM tabular corretamente"
    
    This reverts commit 3e8ba5e05054ba3eb90554d8096c1d000941d75a.

commit 3e8ba5e05054ba3eb90554d8096c1d000941d75a
Author: Fernando Henrique <liquuid@leona.localdomain>
Date:   Wed Feb 10 21:27:36 2010 -0200

    adicionando configuações para o VIM tabular corretamente

commit 29c2ef399ce8d3d707ca34bc2b4f6acdfd661fc8
Author: Fernando Henrique <liquuid@leona.localdomain>
Date:   Tue Feb 9 13:24:18 2010 -0200

    Documentacao e Licenciamento, e versao final para o CCJ

commit c3cfbebbae609350d11f282d7e49a90ad846c92d
Author: Fernando Henrique <liquuid@leona.localdomain>
Date:   Tue Feb 9 08:46:53 2010 -0200

    corretor de cadastros automatizado para o CCJ

commit 00bb8823740f5fad5dcf3977ee28f9f6ba7f9047
Author: Fernando Henrique <liquuid@leona.localdomain>
Date:   Tue Feb 9 08:46:28 2010 -0200

    validadador de datas de nascimento

commit ab3153701625dcca6060aeb06e5623406df97bd3
Author: Fernando Henrique <liquuid@leona.localdomain>
Date:   Mon Feb 8 18:26:11 2010 -0200

    Agora o pylan importa a base de dados de arquivos CSV

commit 5a2d301e38611f154e1881cb973b6824b6bbda6a
Author: Fernando Henrique <liquuid@leona.localdomain>
Date:   Mon Feb 8 15:19:46 2010 -0200

    corrigindo erro de sintaxe primario

commit 22ef575f7f7ce1cad1629a27a37fd6dfeca4f404
Author: Fernando Henrique <liquuid@leona.localdomain>
Date:   Mon Feb 8 12:44:59 2010 -0200

    Backup da lista de usuários para CSV, funcao para checar validade dos forms

commit 1cf4556af0281cd55b2582ecb0f3f58040b8d51b
Author: Fernando Henrique <liquuid@liquuid.net>
Date:   Thu Feb 4 15:51:38 2010 -0200

    alguma docmuentacao

commit d80f223bbaaf66fdde99b55cfa79aee6467c5f9d
Author: Fernando Henrique <liquuid@liquuid.net>
Date:   Wed Feb 3 18:23:25 2010 -0200

    agora a idade não pode ser setada de forma incorreta

commit 4fa288927eb1849fc92baa238792adaddf5d10c2
Author: Fernando Henrique <liquuid@liquuid.net>
Date:   Tue Feb 2 08:35:46 2010 -0200

    sistema de avisos funcional, e limpeza apos termino da sessao

commit 86deb1c4f9a03678a08173e34e5f1d7af0026fe1
Author: Fernando Henrique <liquuid@liquuid.net>
Date:   Mon Feb 1 10:40:04 2010 -0200

    aviso de 5 minutos faltantes

commit 15d776361fd742aea2bec1af48dcafd61f343ac7
Author: Fernando Henrique <liquuid@liquuid.net>
Date:   Tue Jan 26 09:19:23 2010 -0200

    dialogo informando que o tempo esta acabando

commit ecb003ed7e04d5efbcb1b788d5154bc69f9412d6
Author: liquuid <liquuid@vice.(none)>
Date:   Sun Jan 10 19:19:01 2010 -0200

    otimizações no db

commit a3f8ffaaea752895b740f8b7b87740368cc16adc
Author: liquuid <liquuid@vice.(none)>
Date:   Mon Dec 7 11:44:42 2009 -0200

    tirando as estatísticas desse release

commit 55cc0caeb504e8a8d01cbe86958129acf96fc909
Author: liquuid <liquuid@vice.(none)>
Date:   Mon Dec 7 06:13:26 2009 -0200

    Limpando variaveis nao usadas

commit d7380c5cf59fd57615e9a6a464e0f462e55a3fb3
Author: liquuid <liquuid@vice.(none)>
Date:   Sun Dec 6 21:05:16 2009 -0200

    sex pie com legendas melhores e números absolutos

commit 8e3c05b78a2012841906e9402b6e9b851e91d096
Author: liquuid <liquuid@vice.(none)>
Date:   Fri Dec 4 18:02:27 2009 -0200

    Logs com sexo, arquivos CSV salvos com sucesso

commit 83f1ebde7e73c5c513920ecabe1c728676710a5f
Author: liquuid <liquuid@vice.(none)>
Date:   Mon Nov 30 19:58:29 2009 -0200

    logs por mes rolando

commit 826db30d86d62c63b923b1305ee30f887516dab6
Author: liquuid <liquuid@vice.(none)>
Date:   Mon Nov 30 05:55:27 2009 -0200

    histogramas programados

commit e81f78ea28b482312385961b3ec7712c89f6af8c
Author: liquuid <liquuid@vice.(none)>
Date:   Fri Nov 27 19:02:08 2009 -0200

    autocompletion mais inteligente, conectado com a busca diretamente

commit 43e1f1be799e8ee997557cfbde7a0d90347162d3
Author: liquuid <liquuid@vice.(none)>
Date:   Fri Nov 27 18:20:05 2009 -0200

    idade calculada no formulario e logs

commit 8c0382b1215ec5de24dff0643f0a0371b6fda1b0
Author: liquuid <liquuid@vice.(none)>
Date:   Fri Nov 27 13:49:39 2009 -0200

    correções no alinhamento

commit cb915154049d0a8e916dc15c2f273cce883ae3be
Author: liquuid <liquuid@vice.(none)>
Date:   Mon Nov 23 08:39:52 2009 -0200

    Timer paralelo no pylan-controler

commit 223be3618e8a0c7a7db88d1035bac4c42de0612e
Author: liquuid <liquuid@vice.(none)>
Date:   Mon Nov 23 06:24:52 2009 -0200

    mais limpeza no cõdigo

commit 820df5c81b4212effa6b3d16a394ac9e877b5124
Author: liquuid <liquuid@vice.(none)>
Date:   Mon Nov 23 06:21:34 2009 -0200

    Alguma documentação e mais limpeza no cõdigo

commit c31d60457e8570d491b5241cabc64585cdb36408
Author: liquuid <liquuid@vice.(none)>
Date:   Mon Nov 23 05:49:17 2009 -0200

    Refatoração e simplificação do código do pylan-controler, filtros de segurança e maior latência na comunicação com o servidor

commit f583bfa25cb2baa571d78ee07ecb317b8d2b6cbb
Author: liquuid <liquuid@vice.(none)>
Date:   Sat Nov 21 22:47:41 2009 -0200

    logs aparecendo na aba de logs :)

commit 43376fd087ed628d0103932ea86d8e9db84206ea
Author: liquuid <liquuid@vice.(none)>
Date:   Sat Nov 21 15:03:23 2009 -0200

    Search completion funcionando

commit eb85d5e8ce9183b83d5ef147b1240ea0fe690cba
Author: liquuid <liquuid@vice.(none)>
Date:   Sat Nov 21 14:15:15 2009 -0200

    Titulo para a janela do pylan

commit 9c1c8dbc4b85db429b0753691068350a837b460a
Author: liquuid <liquuid@vice.(none)>
Date:   Sat Nov 21 11:23:27 2009 -0200

    fechando arquivos abertos pelo pylanrc

commit f96beb364f050d698600dd95afc77a221e80f837
Author: liquuid <liquuid@vice.(none)>
Date:   Sat Nov 21 11:10:49 2009 -0200


limpadb.py
#!/usr/bin/env python
# _*_ coding: UTF-8 _*_
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Library General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
#
# Copyright 2010 Fernando Henrique R. Silva (liquuid@gmail.com)
# http://hyddro.com



import sqlite3
import os
connection = sqlite3.connect(os.path.expanduser('~/')+'.pylandb.sqlite3')
cur = connection.cursor()
# cur.execute('CREATE TABLE users(id INTEGER PRIMARY KEY,name VARCHAR,gender NUMBER,birthday VARCHAR,grad NUMBER,address VARCHAR,zip VARCHAR,phone VARCHAR,email VARCHAR)')
cur.execute('SELECT id FROM users;')
list = cur.fetchall()

for i in  list:
	cur.execute('SELECT * FROM users where id="'+str(i[0])+'";')
	resu = cur.fetchall()[0]
	for j in xrange(9):
		try:
			if str(resu[j]).count(',') > 0:
				if j == 5:
					cur.execute('replace into users (id,address) values("'+str(resu[0])+'","'+resu[j].replace(',','%')+'")')
					print resu[j]
				elif j == 8:
					cur.execute("replace into users (id,email) values('"+str(resu[0])+"','"+resu[j].replace(',','')+"')")
					print resu[j]

		except UnicodeEncodeError:
				#print resu
				pass
	if resu[3] and len(resu[3]) < 10:
		print "@"+resu[3]

	if resu[3] and len(resu[3]) < 10:
		bd = resu[3].replace('.','/')	
		bd =bd.replace('jan','01').replace('fev','02').replace('mar','03').replace('abr','04').replace('mai','05').replace('jun','06').replace('jul','07').replace('ago','08').replace('set','09').replace('out','10').replace('nov','11').replace('dez','12')
		bdt = bd.split('/')		
		
		if len(bdt) == 3 and bdt[2] == '':
			continue

		if len(bdt) == 3 and bdt[2] != '':
			if len(bdt[1]) == 3:
				bdt[1] = bdt[1][0:2]			
				print bdt[1]

			if len(bdt[2]) == 2:
				bdt[2]= "19"+str(bdt[2])
		
			if len(bdt[0]) < 2:
				bdt[0]= "0"+str(bdt[0])
			if len(bdt[1]) < 2:
				bdt[1] = '0'+bdt[1]
			if int(bdt[0]) > 31:
				bdt[0] = "01"
			if len(bdt[2]) == 3:
				if int(bdt[2]) > 110 and int(bdt[2]) < 200:
					bdt[2] = "19"+bdt[2][1:3]
					
					  
			bd = bdt[0]+'/'+bdt[1]+'/'+bdt[2]
		
			if bd.count('/') == 0 and len(bd) == 8:
				bd = str(bd[0:2])+'/'+str(bd[2:4])+'/'+bd[4:10]
		if len(bdt) == 2:
			if len(bdt[1]) == 6:
				bd = str(bdt[0])+'/'+bdt[1][0:2]+'/'+bdt[1][2:6]
			if len(bdt[0]) == 4:
				bd = str(bdt[0][0:2])+'/'+bdt[0][2:4]+'/'+bdt[1]
				print "--"+bd

		if len(bdt) == 1:
			bd = str(bdt[0][0:2])+'/'+str(bdt[0][2:4])+'/'+bdt[0][4:10]

		if len(bd) == 10:
			cur.execute('replace into users (id,birthday) values("'+str(resu[0])+'","'+bd+'")')
			print resu[0],bd
	
	if resu[3] and len(resu[3]) < 10 :
		print resu[3]	
cur.execute('SELECT id FROM log;')
list = cur.fetchall()

for i in  list:
	cur.execute('SELECT * FROM log where id="'+str(i[0])+'";')

	resu = cur.fetchall()[0]
	if resu[7] and len(resu[7]) < 10:
		print "@"+resu[7]

	if resu[7] and len(resu[7]) < 10:
		bd = resu[7].replace('.','/')	
		bd =bd.replace('jan','01').replace('fev','02').replace('mar','03').replace('abr','04').replace('mai','05').replace('jun','06').replace('jul','07').replace('ago','08').replace('set','09').replace('out','10').replace('nov','11').replace('dez','12')
		bdt = bd.split('/')		
		
		if len(bdt) == 3 and bdt[2] == '':
			continue

		if len(bdt) == 3 and bdt[2] != '':
			if len(bdt[1]) == 3:
				bdt[1] = bdt[1][0:2]			
				print bdt[1]

			if len(bdt[2]) == 2:
				bdt[2]= "19"+str(bdt[2])
		
			if len(bdt[0]) < 2:
				bdt[0]= "0"+str(bdt[0])
			if len(bdt[1]) < 2:
				bdt[1] = '0'+bdt[1]
			if int(bdt[0]) > 31:
				bdt[0] = "01"
			if len(bdt[2]) == 3:
				if int(bdt[2]) > 110 and int(bdt[2]) < 200:
					bdt[2] = "19"+bdt[2][1:3]
					
					  
			bd = bdt[0]+'/'+bdt[1]+'/'+bdt[2]
		
			if bd.count('/') == 0 and len(bd) == 8:
				bd = str(bd[0:2])+'/'+str(bd[2:4])+'/'+bd[4:10]
		if len(bdt) == 2:
			if len(bdt[1]) == 6:
				bd = str(bdt[0])+'/'+bdt[1][0:2]+'/'+bdt[1][2:6]
			if len(bdt[0]) == 4:
				bd = str(bdt[0][0:2])+'/'+bdt[0][2:4]+'/'+bdt[1]
				print "--"+bd

		if len(bdt) == 1:
			bd = str(bdt[0][0:2])+'/'+str(bdt[0][2:4])+'/'+bdt[0][4:10]

		if len(bd) == 10:
			cur.execute('replace into log (id,birthday) values("'+str(resu[0])+'","'+bd+'")')
			print resu[0],bd
	
	if resu[7] and len(resu[7]) < 10 :
		print resu[7]	


connection.commit() 
